installation-bundle/src/InstallTool.php

```diff
@@ -136,6 +136,15 @@ public function setConnection(Connection $connection)
      */
     public function canConnectToDatabase($name)
     {
+        // Return if there is a working database connection already
+        try {
+            $this->connection->connect();
+            $this->connection->query('SHOW TABLES');
+
+            return true;
+        } catch (\Exception $e) {
+        }
+
```

Comment:
wouldn't this break if `$this->connection` is `null` (checked on the next line)?

Reply:
The PR is an exact backport of a2c25b4163ceecb0dfdf6db5fc7a9c47da1a3b4d, which we have added in Contao 4.6. 🤷‍♂

Reply:
Why should it break? It's wrapped in a try/catch block?